### PR TITLE
SV-522 epao apply to new standard

### PR DESCRIPTION
**Pull request by StuartReeve102**

### CODE REVIEW

File: src/SFA.DAS.AssessorService.Web/Views/Application/StandardApplications.cshtml [hunk 1]
```diff
@@ -11,6 +11,9 @@
             <h1 class="govuk-heading-xl">
                 Select application
             </h1>
+            <form asp-action="StartApplication" asp-controller="Application" method="post">
+                <button class="govuk-button" type="submit">Start Application</button>
```
Comment (CodeWolf86) [inline src/SFA.DAS.AssessorService.Web/Views/Application/StandardApplications.cshtml:15]: I think on the prototype this has changed to "Start New Application"

File: src/SFA.DAS.AssessorService.Web/Views/Application/StandardApplications.cshtml [hunk 2]
```diff
@@ -26,23 +29,24 @@
...
-                            <td class="govuk-table__cell govuk-table__cell--numeric">
-                                <a class="govuk-link" href="@Url.Action("SequenceSignPost", "Application", new { Id = application.Id })">View</a>
-                            </td>
-                        </tr>
+                            <tr>
+                                <th class="govuk-table__header" scope="row">
+                                    @application.ApplyData?.Apply?.StandardName
+                                    @if (application.ApplyData?.Apply?.Versions != null && application.ApplyData?.Apply?.Versions.Count > 0)
+                                    {
+                                        <p class="govuk-body">Version @application.ApplyData?.Apply?.Versions[0].</p>
```
Comment (CodeWolf86) [inline src/SFA.DAS.AssessorService.Web/Views/Application/StandardApplications.cshtml:37]: Just trying to find the page for this to match to the prototype, if there is more than one version, do we not want to list the versions rather than show the latest?

File: src/SFA.DAS.AssessorService.Web/Views/Application/Standard/ConfirmStandard.cshtml [hunk 3]
```diff
@@ -75,12 +99,12 @@
                             <div class="govuk-checkboxes__item">
                                 <input class="govuk-checkboxes__input" asp-for="@Model.IsConfirmed" name="IsConfirmed" type="checkbox">
                                 <label class="govuk-label govuk-checkboxes__label" for="IsConfirmed">
-                                    By continuing with this application you are confirming you have <a href="@Model.SelectedStandard.StandardData.AssessmentPlanUrl" target="_blank" class="govuk-link">read the assessment plan</a> (PDF).
+                                    By continuing, you confirm you have read and understood the relevant assessment plan(s) for <a href="@Model.SelectedStandard?.StandardPageUrl" target="_blank" class="govuk-link">this standard.</a>
```
Comment (CodeWolf86) [inline src/SFA.DAS.AssessorService.Web/Views/Application/Standard/ConfirmStandard.cshtml:102]: This is probably a quirk of the HTML, but when testing in IE, I wasn't able to click the link to view the standard page url. It instead ticked/unticked the tickbox, I'm assuming it's becaue it's wrapped in the label....

File: src/SFA.DAS.AssessorService.Web/ViewModels/Apply/StandardVersionViewModel.cs [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+namespace SFA.DAS.AssessorService.Web.ViewModels.Apply
+{
+    public class StandardVersionViewModel
+    {
+        public Guid Id { get; set; }
+
+        public string StandardToFind { get; set; }
+
+        public string StandardCode { get; set; }
```
Comment (CodeWolf86) [inline src/SFA.DAS.AssessorService.Web/ViewModels/Apply/StandardVersionViewModel.cs:13]: I think `StandardCode` should be renamed `StandardReference` here.

In the StandardVersion model, we try to separate by LarsCode being the integer value, and IFateReference being the ST00ZZ value. 

In existing code bases they were distinguished via StandardCode = LarsCode and StandardReference = IFateReference. So if we're sticking with existing terminology, StandardRef is correct here ( as i believe it gets populate with an ST value ? )

File: src/SFA.DAS.AssessorService.Web/Controllers/Apply/StandardController.cs [hunk 5]
```diff
@@ -46,42 +49,59 @@ public async Task<IActionResult> Search(StandardViewModel model)
                 return RedirectToAction(nameof(Index), new { model.Id });
             }
 
-            var standards = await _apiClient.GetStandards();
+            var standards = await _standardVersionApiClient.GetAllStandardVersions();
```
Comment (CodeWolf86) [inline src/SFA.DAS.AssessorService.Web/Controllers/Apply/StandardController.cs:52]: I would be tempted to add a GetLatestStandardVersions on the StandardVersionController to extend it. What do you think?

File: src/SFA.DAS.AssessorService.Web/Controllers/Apply/StandardController.cs [hunk 6]
```diff
@@ -46,42 +49,59 @@ public async Task<IActionResult> Search(StandardViewModel model)
...
+            var standardViewModel = new StandardVersionViewModel { Id = id, StandardCode = standardCode };
             if (!CanUpdateApplicationAsync(application))
             {
                 return RedirectToAction("Applications", "Application");
             }
 
-            var standards = await _apiClient.GetStandards();
-            standardViewModel.SelectedStandard = standards.FirstOrDefault(s => s.StandardId == standardCode);
-            standardViewModel.ApplicationStatus = await ApplicationStandardStatus(application, standardCode);
+            var standards = (await _standardVersionApiClient.GetAllStandardVersions())
```
Comment (CodeWolf86) [inline src/SFA.DAS.AssessorService.Web/Controllers/Apply/StandardController.cs:78]: I'd also be tempted here to add a GetStandardVersionsByIFateReferenceNumber, We already have a similiar but by LarsCode so would simplify the call and logic, thoughts?

File: src/SFA.DAS.AssessorService.Web/Controllers/Apply/StandardController.cs [hunk 7]
```diff
@@ -46,42 +49,59 @@ public async Task<IActionResult> Search(StandardViewModel model)
...
+                    })
+                .ToList();
 
             return View("~/Views/Application/Standard/FindStandardResults.cshtml", model);
         }
 
         [HttpGet("standard/{id}/confirm-standard/{standardCode}")]
         [ApplicationAuthorize(routeId: "Id")]
-        public async Task<IActionResult> ConfirmStandard(Guid id, int standardCode)
+        public async Task<IActionResult> ConfirmStandard(Guid id, string standardCode)
```
Comment (CodeWolf86) [inline src/SFA.DAS.AssessorService.Web/Controllers/Apply/StandardController.cs:69]: I'd tweak the standardCode to StandardReference/IfateReferenceNumber now it's using the STxxx rather than lars code

**Review by CodeWolf86 — CHANGES_REQUESTED** [review]

**Comment by CodeWolf86** [comment]
@StuartReeve102  It looks good, i've ran through the process and applied for specific versions so it all hangs together. I've added some comments for you to have a look at, and happy to talk them through as discussion points. Only thing missing is some unit tests on the StandardController as while none exist at the moment we're trying to improve as we go.

File: src/SFA.DAS.AssessorService.Web/web.config [hunk 8]
```diff
@@ -6,7 +6,7 @@
         <requestFiltering allowDoubleEscaping="true" />
       </security>
       <handlers>
-        <add name="aspNetCore" path="*" verb="*" modules="AspNetCoreModule" resourceType="Unspecified" />
+        <add name="aspNetCore" path="*" verb="*" modules="AspNetCoreModuleV2" resourceType="Unspecified" />
```
[the remark this answers is not in the document]
Comment (StuartReeve102) [reply src/SFA.DAS.AssessorService.Web/web.config:9]: No looks like an automatic change by VS - Fixed

File: src/SFA.DAS.AssessorService.Api.Types/Models/Standards/StandardVersion.cs [hunk 9]
```diff
@@ -13,7 +13,9 @@ public class StandardVersion
         public int LarsCode { get; set; }
         public int Level { get; set; }
         public DateTime EffectiveFrom { get; set; }
+        public DateTime? EffectiveTo { get; set; }
```
[the remark this answers is not in the document]
Comment (StuartReeve102) [reply src/SFA.DAS.AssessorService.Api.Types/Models/Standards/StandardVersion.cs:16]: Both as now Nullable inline with the DB schema - Fixed

File: src/SFA.DAS.AssessorService.Web/Views/Application/StandardApplications.cshtml (hunk 1 above)
Comment (StuartReeve102) [reply src/SFA.DAS.AssessorService.Web/Views/Application/StandardApplications.cshtml:15]: Its does - Fixed
NB - we need a way of highlighting prototype changes

File: src/SFA.DAS.AssessorService.Web/Views/Application/StandardApplications.cshtml (hunk 2 above)
Comment (StuartReeve102) [reply src/SFA.DAS.AssessorService.Web/Views/Application/StandardApplications.cshtml:37]: Changed to list the versions

File: src/SFA.DAS.AssessorService.Web/Views/Application/Standard/ConfirmStandard.cshtml (hunk 3 above)
Comment (StuartReeve102) [reply src/SFA.DAS.AssessorService.Web/Views/Application/Standard/ConfirmStandard.cshtml:102]: The url import was missing from the PR hence the link wasnt woking. Fixed now.

File: src/SFA.DAS.AssessorService.Web/ViewModels/Apply/StandardVersionViewModel.cs (hunk 4 above)
Comment (StuartReeve102) [reply src/SFA.DAS.AssessorService.Web/ViewModels/Apply/StandardVersionViewModel.cs:13]: Changed to StandardReference

File: src/SFA.DAS.AssessorService.Web/Controllers/Apply/StandardController.cs (hunk 5 above)
Comment (StuartReeve102) [reply src/SFA.DAS.AssessorService.Web/Controllers/Apply/StandardController.cs:52]: I assume you mean to move the grouping to SQL. Can we leave it for now and make the change in a future story, as I think there is definitely a requirement for this functionality?

File: src/SFA.DAS.AssessorService.Web/Controllers/Apply/StandardController.cs (hunk 6 above)
Comment (StuartReeve102) [reply src/SFA.DAS.AssessorService.Web/Controllers/Apply/StandardController.cs:78]: See previous note.

File: src/SFA.DAS.AssessorService.Data.IntegrationTests/SFA.DAS.AssessorService.Data.IntegrationTests.csproj [hunk 10]
```diff
@@ -31,6 +31,9 @@
   </ItemGroup>
 
   <ItemGroup>
+    <None Update="connectionStrings.Local.json">
```
[the remark this answers is not in the document]
Comment (StuartReeve102) [reply src/SFA.DAS.AssessorService.Data.IntegrationTests/SFA.DAS.AssessorService.Data.IntegrationTests.csproj]: Fixed

File: src/SFA.DAS.AssessorService.Web/Controllers/Apply/StandardController.cs (hunk 5 above)
Comment (CodeWolf86) [reply src/SFA.DAS.AssessorService.Web/Controllers/Apply/StandardController.cs:56]: Added Sub task to SV-564 https://skillsfundingagency.atlassian.net/browse/SV-822

File: src/SFA.DAS.AssessorService.Web/Controllers/Apply/StandardController.cs (hunk 6 above)
Comment (CodeWolf86) [reply src/SFA.DAS.AssessorService.Web/Controllers/Apply/StandardController.cs:84]: Added Sub task to SV-564 https://skillsfundingagency.atlassian.net/browse/SV-822

**Review by CodeWolf86 — APPROVED** [review]
